streamer/src/nonblocking/quic.rs

```diff
@@ -218,7 +243,37 @@ async fn run_server(
         }
 
         if let Ok(Some(connection)) = timeout_connection {
-            info!("Got a connection {:?}", connection.remote_address());
+            let remote_address = connection.remote_address();
+
+            // first check overall connection rate limit:
+            if !overall_connection_rate_limiter.is_allowed() {
+                debug!(
+                    "Reject connection from {:?} -- total rate limiting exceeded",
+                    remote_address.ip()
+                );
+                stats
+                    .connection_throttled_across_all
+                    .fetch_add(1, Ordering::Relaxed);
+                continue;
+            }
+
+            if rate_limiter.len() > CONNECITON_RATE_LIMITER_CLEANUP_SIZE_THRESHOLD {
```

Comment:
would not this check cause a lot of troubles in the worst case when we would have more than threshold entries? every connection would cause an iteration of over 100k entries
maybe `retain_recent` should not be called more often than 10ms for example?

Reply:
Am I right that we might be in situation when we have > 100k IP addresses in the map and all of these connections are recent (<60s) so we will go over this dashmap over and over without removing enough connections to be below 100k? Or this situation is only hypothetical because having ~100k IP addresses in the last minute is almost impossible event?

Reply:
> Am I right that we might be in situation when we have > 100k IP addresses in the map and all of these connections are recent (<60s) so we will go over this dashmap over and over without removing enough connections to be below 100k?

Yes, this looks correct, except that we also have a check above that does not allow more than 150k connections per 60s (based on TOTAL_CONNECTIONS_PER_SECOND = 2500).

> Or this situation is only hypothetical because having ~100k IP addresses in the last minute is almost impossible event?

Not necessary to have 100k IP's, node allows multiple connections from 1 IP, 8 iirc? Still sounds hard but possible?